workers/ohsome_quality_analyst/indicators/mapping_saturation/indicator.py

```diff
@@ -35,162 +36,177 @@ def __init__(
-            self.no_data = True
-        elif self.values[-1] == 0:
-            self.deleted_data = True
-        else:
-            self.values_normalized = [value / max_value for value in self.values]
+        for item in query_results["result"]:
+            self.values.append(item["value"])
+            self.timestamps.append(isoparse(item["timestamp"]))
 
     def calculate(self) -> None:
-        """Calculate the growth rate and saturation level within the last 3 years."""
-        if self.no_data:
-            self.result.description = "No features were mapped in this region."
+        # Latest timestamp of ohsome API results
+        self.result.timestamp_osm = self.timestamps[-1]
+        edge_case_description = self.check_edge_cases()
+        if edge_case_description:
+            logging.info("Edge case is present. Skipping indicator calculation.")
+            self.result.description = edge_case_description
             return
-        if self.deleted_data:
-            self.result.description = (
-                "All mapped features in this region have been since deleted."
-            )
+        xdata = np.array(range(len(self.timestamps)))
+        fitted_models = []
+        for model in (
+            models.Sigmoid,
+            models.SSlogis,
+            models.SSdoubleS,
+            models.SSfpl,
+            models.SSasymp,
+            models.SSmicmen,
+        ):
+            logging.info("Run {}".format(model.name))
+            try:
+                fitted_models.append(model(xdata=xdata, ydata=np.array(self.values)))
+            except RRuntimeError as error:
+                logging.warning(
+                    "Could not run model {0} due to RRuntimeError: {1}".format(
+                        model.name, error
+                    )
+                )
+                continue
+        self.fitted_models = self.select_models(fitted_models)
+        if not self.fitted_models:
+            logging.info("No model has been run successfully.")
             return
-        # prepare the data
-
-        # get y values fot best fitting sigmoid curve, with these y the
-        # saturation will be calculated
-        sigmoid_curve = sigmoidCurve()
-        # TODO: Change the dict to three parameters
-        ydata_for_sat = sigmoid_curve.getBestFittingCurve(
-            {
-                "timestamps": self.timestamps,
-                "results": self.values,
-                "results_normalized": self.values_normalized,
-            }
+        self.best_fit = min(self.fitted_models, key=lambda m: m.mae)
+        logging.info("Best fitting model: " + self.best_fit.name)
+        # Calculate saturation: The slope of the last 3 years of the fitted curve
+        self.saturation = (
+            np.interp(xdata[-36], xdata, self.best_fit.fitted_values)
+        ) / (np.interp(xdata[-1], xdata, self.best_fit.fitted_values))
+        self.result.value = self.values[-1] / self.best_fit.asymptote  # Completeness
```

Comment:
I added a new attribute called `latest_value`. I do not want to call it latest_feature_count or similar since it is not always a count (e.g. road length)